File: app/src/main/java/org/thoughtcrime/securesms/service/ContactsSyncAdapterService.java
package org.thoughtcrime.securesms.service;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import androidx.annotation.Nullable;

import org.thoughtcrime.securesms.contacts.ContactsSyncAdapter;

public class ContactsSyncAdapterService extends Service {

  private static ContactsSyncAdapter syncAdapter;

  @Override
  public synchronized void onCreate() {
    if (syncAdapter == null) {
      syncAdapter = new ContactsSyncAdapter(this, true);
    }
  }

  @Nullable
  @Override
  public IBinder onBind(Intent intent) {
    return syncAdapter.getSyncAdapterBinder();
  }
}

File: demo/contacts/src/main/java/org/signal/contactstest/ContactsSyncAdapterService.kt
package org.signal.contactstest

import android.app.Service
import android.content.Intent
import android.os.IBinder

class ContactsSyncAdapterService : Service() {
  @Synchronized
  override fun onCreate() {
    if (syncAdapter == null) {
      syncAdapter = ContactsSyncAdapter(this, true)
    }
  }

  override fun onBind(intent: Intent): IBinder? {
    return syncAdapter!!.syncAdapterBinder
  }

  companion object {
    private var syncAdapter: ContactsSyncAdapter? = null
  }
}
